### DISCUSSION TITLE
Plugin Store: UI Enhancement - Buy Button

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#8649] [Ideas] [answered]
### Description
Personally I think moving the “buy/add to cart” button to the top right of the individual plugin page would be a great enhancement. Having the buy button in a predictable place that doesn’t change from plugin page to plugin page is a much better user experience.

Having to scroll each individual plugin page to find the buy button is harder than it should be. If a plugin developer provides a thorough description for their plugin (which is great), it pushes the buy button further down the page (below the fold) which is not so great for someone who has already made up their mind to buy.

Furthermore it would be awesome if you could add plugins to the cart directly from the plugin store listing pages. I just added 20+ plugins to my cart and it took way longer than it should have if any of the two enhancements above had been implemented.

**Comment by brandonkelly** [comment]
It used to be up in the top-right, but we moved it down below in Craft 3.1 specifically so it’s in a consistent place between single-edition plugins and multi-edition plugins.

That said, having a shortcut button at the top-right that just scrolls the page down to the edition boxes wouldn’t be a bad idea.

**Comment by richhayler** [comment]
As part of your suggested solution it would be great to consider that on a plugin listing page (e.g. plugin category), there was a buy button listed for each plugin and when clicked that took you to the individual plugin page and scrolled to the buy options. I’m considering people who are very familiar with the plugin store and specific plugins and know what they want to buy. How to make that super easy to add a plugin to your cart and complete the purchase.

**Comment by richhayler** [comment]
I just added 20+ plugins to my cart for a new project and it was harder that it should be. I ended up opening multiple browser tabs for each plugin page to avoid using the browser back button to take me back to the page on which there were additional plugins I wanted to add to my cart.

**Comment by brandonkelly** [comment]
We’re working on making it possible to fave plugins, and I could see us adding Add to Cart buttons the “My Favorites” page, but not on the main plugin listings – that would be too busy.

**Comment by richhayler** [comment]
That would work for me.

**Comment by jsunsawyer** [comment]
I also think it'd be convenient to directly install plugins from the list view instead of being required to go into each, wait for content to load (5-6 seconds watching a spinner for me), click an anchor link (currently, scroll down until you get past all of the author's notes), then click install.

An anchor link might help a bit, but it's really just replacing the "scroll down" step.

It's just not a pleasant experience using the plugin store right now.

**Comment by brandonkelly** [comment, marked as answer]
We significantly improved the plugin detail page layout in Craft 4, including moving the buy button(s) to the top of the sidebar.
